app/controllers/tweet_controller.rb

```diff
@@ -0,0 +1,62 @@
+class TweetController < ApplicationController
+
+  get '/tweets' do
+    if loggedin?
+       erb :'/tweets/tweets'
+     else
+       redirect '/login'
+    end
+  end
+
+  get '/tweets/new' do
+    if loggedin?
+      erb :create_tweet
+    else
+      redirect '/login'
+    end
+  end
+
+  get '/tweets/:id/edit' do
+    if loggedin?
+      @tweet = Tweet.find_by(user_id: params[:id])
+      erb :'/tweets/edit_tweet'
+    else
+      redirect '/login'
+    end
+  end
+
+  delete '/delete' do
+    tweet = Tweet.find_by(user_id: session[:id])
+    tweet.delete
+    redirect '/tweets'
+  end
+  patch '/edit' do
+    if params['content'] != ""
+     tweet = Tweet.find_by(user_id: session[:id])
+     tweet.content = params['content']
+     tweet.save
+   else
+     redirect "tweets/#{session[:id]}/edit"
+   end
+  end
+
+  post '/tweets/' do
+    if params['content'] != ""
```

Comment:
I think you could use [ActiveRecord validations](http://guides.rubyonrails.org/active_record_validations.html) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.